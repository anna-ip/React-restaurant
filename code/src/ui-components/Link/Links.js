import React from "react";
import { Link } from "react-router-dom";
import styles from "./Links.module.scss";

const Links = ({ to, text, className }) => {
  return (
    <Link to={to} text={text} className={className + " " + styles.link}>
      {text}
    </Link>
  );
};

export default Links;
