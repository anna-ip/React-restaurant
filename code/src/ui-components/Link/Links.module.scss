@import '../../styles/_variables.scss';
.link {
  cursor: pointer;
  border: none;
  background: inherit;
  color: $red-color;
  text-decoration: none;

    &:focus,
    &:hover {
      transform: scale(1.1);
      text-decoration: underline;
    }
}