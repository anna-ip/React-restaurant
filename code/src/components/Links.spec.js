import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import Link from './Links.js'

//mock the react-router to be able to snapshot the Link component
jest.mock('react-router-dom')

it('<Link /> renders "null"', () => {
  //not sure this is the correct way?
  Router.mockReturnValue({})

  const tree = renderer
    .create(<Link to="http://localhost:3000">Go to home page</Link>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
