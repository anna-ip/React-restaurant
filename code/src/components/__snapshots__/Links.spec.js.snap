// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> renders "null" 1`] = `null`;
